provers/debug/prover.go

```
@@ -0,0 +1,157 @@
+package debug
+
+import (
+	"context"
+	fmt "fmt"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	clienttypes "github.com/cosmos/ibc-go/v8/modules/core/02-client/types"
+	"github.com/cosmos/ibc-go/v8/modules/core/exported"
+
+	//mocktypes "github.com/datachainlab/ibc-mock-client/modules/light-clients/xx-mock/types"
+	"github.com/hyperledger-labs/yui-relayer/core"
+	"github.com/hyperledger-labs/yui-relayer/log"
+)
+
+func debugFakeLost(ctx context.Context, chain core.Chain, queryHeight exported.Height) error {
+	if val, ok := os.LookupEnv("DEBUG_RELAYER_MISSING_TRIE_NODE_HEIGHT_PROVER"); ok {
+		s := strings.Split(val, " ")
+		if len(s) != 2 {
+			fmt.Printf("malformed DEBUG_RELAYER_MISSING_TRIE_NODE_HEIGHT_PROVER: Expected format: <chainid> <height threshold>\n")
+			return nil
+		}
+		if s[0] == chain.ChainID() {
+			threshold, err := strconv.Atoi(s[1])
+			if err != nil {
+				fmt.Printf("malformed DEBUG_RELAYER_MISSING_TRIE_NODE_HEIGHT_PROVER: %v\n", err)
+				return nil
+			}
```

We should move L28-L32 to the outside of the `if s[0] == chain.ChainID()` block.
When the `DEBUG_RELAYER_MISSING_TRIE_NODE_HEIGHT` variable is set but `s[1]` is malformed, this function should return an error here.

---

Changed environment variable name include chain id because it is enabled to set debug against multiple chains.